user-interface/src/staff-assignment/StaffAssignmentRow.test.tsx

```diff
@@ -0,0 +1,165 @@
+    });
+  });
+
+  test('should show "unassigned" if there is not an assigned attorney', async () => {
+    vi.spyOn(Api2, 'getCaseAssignments').mockResolvedValue(
+      buildResponseBodySuccess<CaseAssignment[]>([]),
+    );
+
+    renderWithProps();
+
+    await waitFor(() => {
+      const firstTable = document.querySelector('table');
+      const cols = firstTable?.querySelectorAll('td');
+      expect(cols?.[4]).toHaveTextContent('(unassigned)');
+      const button = cols?.[4].querySelector('button');
+      expect(button).not.toHaveClass(UswdsButtonStyle.Outline);
+      expect(button).toHaveTextContent('Assign');
+      expect(button).toHaveAttribute('title', 'add assignments');
+      expect(button).toHaveAttribute('data-open-modal', 'true');
+    });
+  });
+
+  test('should show assigned attorney names for assigned attorneys', async () => {
+    const assignments = MockData.buildArray(MockData.getAttorneyAssignment, 2);
+    vi.spyOn(Api2, 'getCaseAssignments').mockResolvedValue(
+      buildResponseBodySuccess<CaseAssignment[]>(assignments),
+    );
+
+    renderWithProps();
+
+    await waitFor(() => {
+      const firstTable = document.querySelector('table');
+      const cols = firstTable?.querySelectorAll('td');
+      assignments.forEach((assignment) => {
+        expect(cols?.[4]).toHaveTextContent(assignment.name);
+        const button = cols?.[4].querySelector('button');
+        expect(button).toHaveClass(UswdsButtonStyle.Outline);
+        expect(button).toHaveTextContent('Edit');
+        expect(button).toHaveAttribute('title', 'edit assignments');
+        expect(button).toHaveAttribute('data-open-modal', 'true');
+      });
+    });
+  });
+
+  test('should not show assign/edit button', async () => {
+    const assignments = MockData.buildArray(MockData.getAttorneyAssignment, 2);
+    vi.spyOn(Api2, 'getCaseAssignments').mockResolvedValue(
+      buildResponseBodySuccess<CaseAssignment[]>(assignments),
+    );
+
+    const myCase = { ...bCase, _actions: [] };
+    renderWithProps({ bCase: myCase });
+
+    await waitFor(() => {
+      const firstTable = document.querySelector('table');
+      const cols = firstTable?.querySelectorAll('td');
+      assignments.forEach((assignment) => {
+        expect(cols?.[4]).toHaveTextContent(assignment.name);
+        const button = cols?.[4].querySelector('button');
+        expect(button).not.toBeInTheDocument();
+      });
+    });
+  });
+
+  test('should show error alert when an error is thrown by api.getCaseAssignments', async () => {
+    vi.spyOn(Api2, 'getCaseAssignments').mockRejectedValue('some error');
+    const globalAlertSpy = testingUtilities.spyOnGlobalAlert();
+
+    renderWithProps();
+
+    await waitFor(() => {
+      expect(globalAlertSpy.error).toHaveBeenCalledWith(
+        expect.stringContaining('Could not get staff assignments for case '),
+      );
+    });
+  });
+
+  // TODO: Test useStaffAssignmentRowActions.actions.getCaseAssignments()
```

Comment:
Is this a comment left over from one of the experiments were doing or is this something we can remove?

Reply:
Yes. I cover this in the experiment.